agent/agent.go

```diff
@@ -618,29 +623,37 @@ func (a *Agent) endTaskResponse(tc *taskContext, status string, message string)
 	return detail
 }
 
-func (a *Agent) runPostTaskCommands(ctx context.Context, tc *taskContext) {
+func (a *Agent) runPostTaskCommands(ctx context.Context, tc *taskContext) error {
 	start := time.Now()
 	a.killProcs(ctx, tc, false)
 	defer a.killProcs(ctx, tc, false)
 	tc.logger.Task().Info("Running post-task commands.")
+	opts := runCommandsOptions{}
 	postCtx, cancel := a.withCallbackTimeout(ctx, tc)
 	defer cancel()
 	taskConfig := tc.getTaskConfig()
 	taskGroup, err := taskConfig.GetTaskGroup(tc.taskGroup)
 	if err != nil {
 		tc.logger.Execution().Error(errors.Wrap(err, "error fetching task group for post-task commands"))
-		return
+		return nil
 	}
 	if taskGroup.TeardownTask != nil {
-		err = a.runCommands(postCtx, tc, taskGroup.TeardownTask.List(), runCommandsOptions{})
-		tc.logger.Task().Error(message.WrapError(err, message.Fields{
-			"message": "Error running post-task command.",
-		}))
+		opts.shouldSetupFail = taskGroup.TeardownGroupFailTask
```

Comment:
shouldSetupFail is part of the runCommandsOptions struct.  Currently it is set as flagged to true if pre_error_fails_task is true because this will set SetupGroupFailTask and this property is referring to that. We can introduce a new property in the struct called something like shouldTeardownFail, or we could rename this property to something more generic that can apply to pre and post task commands, like shouldTriggerFailure.